### PR TITLE
⚠️  Bump go 1.19 , CAPI to v1.3.0-beta.1, Ginkgo/v2 and bump related go modules

### PR DESCRIPTION
**Pull request by kashifest**
Also adds a fix for kustomization for darwin.

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
This is testing CAPM3 itself but we could /should test it including the IPAM changes in https://github.com/metal3-io/ip-address-manager/pull/164 as well

File: baremetal/remote/remote_test.go
Comment (lentzi90) [inline baremetal/remote/remote_test.go:75]: This looks a bit weird to me. Is the linter complaining? I can see `calidKubeConfig` being used below so should be fine. Maybe the linter didn't get it?

**Review by lentzi90 — COMMENTED** [review]
Nice job!
A couple of questions below

File: scripts/ci-e2e.sh
Comment (kashifest) [reply scripts/ci-e2e.sh:18]: ow no. There's a test commit which I need to remove. Waiting for review so that, I will push the changes all together.

File: Makefile
Comment (mboukhalfa) [inline Makefile:182]: we might not need `time` and `go test -timeout`

**Review by furkatgofurov7 — COMMENTED** [review]
Can you please also squash your commits?

**Comment by mboukhalfa** [comment]
/retitle 🌱 Bump go 1.19 , CAPI to v1.3.0-beta.1 and bump related go modules

File: Makefile
Comment (kashifest) [reply Makefile:182]: Yes. I dont want to touch that in this PR. I will do a followup.

Comment (kashifest) [reply Makefile:182]: According to @mboukhalfa thats the timeout we have for e2e probably in project-infra. The reason to add this var was because with gingko v2 the default timeout is now 1 hr which is why the PRs were failing previously.

**Comment by kashifest** [comment]
> Can you please also squash your commits?

Yes I will.

File: baremetal/remote/remote_test.go
Comment (kashifest) [reply baremetal/remote/remote_test.go:75]: @furkatgofurov7 do you remember why did I do this, I think you were there when we did this but I have forgotten the reason unfortunately.

File: Makefile
Comment (mboukhalfa) [reply Makefile:182]: Jenkins timeout after 6h, we can use it as default value but keep it parameterized `timeout ?= 6h`

File: baremetal/remote/remote_test.go
Comment (furkatgofurov7) [reply baremetal/remote/remote_test.go:75]: Yes, there was a problem with one of the linters, complaining "declared but not used" kind of error ?

Comment (furkatgofurov7) [reply baremetal/remote/remote_test.go:75]: @kashifest do you still see that error locally without this change? I just checked it in your branch and have not seen this. I even added removed linters back from the golangci.yaml and removed this change thinking one of those were throwing it, but everything went fine. Can you verify it locally as well?

File: .golangci.yaml
Comment (kashifest) [reply .golangci.yaml:45]: I removed them because CAPI is not using these either

**Comment by kashifest** [comment]
> > Can you please also squash your commits?
> 
> Yes I will.

Done

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Comment by mboukhalfa** [comment]
You can also use ⚠️ as ginkgo uplift is breaking changes

**Review by smoshiur1237 — APPROVED** [review]
LGTM

**Comment by lentzi90** [comment]
> You can also use warning as ginkgo uplift is breaking changes

Ginkgo is just for the test code. Generally changes to test code is not considered when deciding if the change is breaking or not, since it will not impact end users.

**Comment by furkatgofurov7** [comment]
> > You can also use warning as ginkgo uplift is breaking changes
> 
> Ginkgo is just for the test code. Generally changes to test code is not considered when deciding if the change is breaking or not, since it will not impact end users.

I would generally agree, to label it as a breaking change (considering CAPI PR labelling https://github.com/kubernetes-sigs/cluster-api/pull/6906). Also, CAPI uplift is considered a breaking change and once this is merged, we would have a minor release from the provider, so +1 for ⚠️ in PR title here and in https://github.com/metal3-io/ip-address-manager/pull/164

**Comment by lentzi90** [comment]
Agreed, the CAPI minor version uplift makes sense as breaking change!

**Comment by furkatgofurov7** [comment]
https://github.com/metal3-io/metal3-dev-env/pull/1097 is merged, triggering the main test

/test-ubuntu-integration-main

**Comment by furkatgofurov7** [comment]
Unrelated CI issue

/test-ubuntu-integration-main

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**
